spec.md

```diff
@@ -1793,6 +1843,42 @@ Condition | gRPC Code | Description | Recovery Behavior
 | Call not implemented | 12 UNIMPLEMENTED | NodeGetInfo call is not implemented by the plugin or disabled in the Plugin's current mode of operation. | Caller MUST NOT retry. Caller MAY call `ControllerGetCapabilities` or `NodeGetCapabilities` to discover Plugin capabilities. |
 
 
+#### `NodeExpandVolume`
```

Comment:
Although I see that this call is for solving the problem that the controller may not be able to resize the filesystem, I'm not sure if this call fits the node service perfectly. Unlike other calls in the node service that serves per-node functionalities, this sounds an "exclusive" call, meaning that we only want to run this call once on some node.

Since the controller plugin is mostly likely to be run on some "node", is it possible to just encapsulate all filesystem changes in the `ControllerExpandVolume`, and let the plugin itself to do whatever necessary under the hood?

Reply:
Or, is it possible to just let `NodeStageVolume` or `NodePublishVolume` lazily perform node-side resize when necessary?

I have to admit this sounds a little hacky, but it's also not natural for the CO to find some random available node to perform an operation that is not related to workload scheduling.

Reply:
It may be possible to roll - node side resizing feature into `NodeStageVolume` or `NodePublishVolume` - but that also means that the case of supporting online resizing will become messier. 

To support online resizing, the CO must be able to call `NodeExpandVolume` on a already "available" volume on the node.  This requirement pretty much rules out `NodeStageVolume` and it seems awkward to call `NodePublishVolume` again for the purpose of resizing the volume(although doable). I think - we should rather be explicit and make separate `NodeExpandVolume` call available so as CO can all it for doing online resizing.